pkg/perf/monitor.go

```diff
@@ -0,0 +1,86 @@
+package perf
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-co-op/gocron"
+	"github.com/gomodule/redigo/redis"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg/utils"
+)
+
+// PerformanceMonitor holds the module data
+type PerformanceMonitor struct {
+	scheduler *gocron.Scheduler
+	pool      redis.Pool
+	tasks     []Task
+}
+
+// NewPerformanceMonitor returns PerformanceMonitor instance
+func NewPerformanceMonitor(redisAddr string) (*PerformanceMonitor, error) {
+	redisPool, err := utils.NewRedisPool(redisAddr)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed creating new redis pool")
+	}
+
+	scheduler := gocron.NewScheduler(time.UTC)
+
+	return &PerformanceMonitor{
+		scheduler: scheduler,
+		pool:      *redisPool,
```

Comment:
You can't do that here, this fails already in CI. redisPool is already a pointer right? and that's is exactly what u need to keep on the PerfomanceMonitor structure

Reply:
The reason this is done because when u do a `deref` (with `*`) on a point value it takes a copy. Pools (at least this one) is not designed to be copied other wise u will end up with 2 pools that don't share internal state. Instead you need always to use a pointer to the same pool hence copying the value is not good.

It's not caught by the compiler but by the CI tests